java/tests/com/google/privacy/differentialprivacy/ConfidenceIntervalGaussianTest.java

```diff
@@ -0,0 +1,289 @@
+package com.google.privacy.differentialprivacy;
+
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.Truth.assertWithMessage;
+import static org.junit.Assert.assertThrows;
+
+import com.google.auto.value.AutoValue;
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import javax.annotation.Nullable;
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Collection;
+
+@RunWith(Enclosed.class)
+public class ConfidenceIntervalGaussianTest {
+    private static final Noise gauss = new GaussianNoise();
+    private static final double TOLERANCE = 1e-7;
+
+    @AutoValue
+    public abstract static class ConfidenceIntervalTestCase implements Serializable {
+        static ConfidenceIntervalTestCase create(
+                double noisedX,
+                int l0Sensitivity,
+                double lInfSensitivity,
+                double epsilon,
+                Double delta,
+                double alpha,
+                ConfidenceInterval expected) {
+            return new AutoValue_ConfidenceIntervalGaussianTest_ConfidenceIntervalTestCase(
+                    noisedX, l0Sensitivity, lInfSensitivity, epsilon, delta, alpha, expected);
+        }
+
+        abstract double noisedX();
+
+        abstract int l0Sensitivity();
+
+        abstract double lInfSensitivity();
+
+        abstract double epsilon();
+
+        abstract Double delta();
+
+        abstract double alpha();
+
+        abstract ConfidenceInterval expected();
+    }
+
+    @RunWith(Parameterized.class)
+    public static class computeConfidenceIntervalDoubleTests {
+        private final ConfidenceIntervalTestCase testCase;
+
+        public computeConfidenceIntervalDoubleTests(ConfidenceIntervalTestCase testCase) {
+            this.testCase = testCase;
+        }
+
+        @Parameterized.Parameters(name = "{index}: = failed")
+        public static Collection<Object> getTestCases() {
```

Comment:
We discussed this with Maria earlier, the main argument was we wanted to do argument checking tests similar to the ones we had in Go and those "Arbitrary" tests. This would probably take 40 or more separate function tests, a lot of repeated code and would be not as readable as this.

I think it makes sense to add those the tests you suggested, that would be even more test cases which would be fine with this approach. We can discuss this with Maria again. Also, if we chose to add them, we need to add them to the Go library to in a separate PR.